homeassistant/components/transmission/sensor.py

```diff
@@ -49,18 +123,29 @@ async def async_setup_entry(
 class TransmissionSensor(SensorEntity):
     """A base class for all Transmission sensors."""
 
-    def __init__(self, tm_client, client_name, sensor_name, sub_type=None):
+    entity_description: TransmissionSensorEntityDescription
+
+    def __init__(
+        self,
+        tm_client: TransmissionClient,
+        client_name: str,
+        entity_description: TransmissionSensorEntityDescription,
+    ) -> None:
         """Initialize the sensor."""
-        self._tm_client: TransmissionClient = tm_client
+        self.entity_description = entity_description
+        self._tm_client = tm_client
         self._client_name = client_name
-        self._name = sensor_name
-        self._sub_type = sub_type
         self._state = None
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, tm_client.config_entry.entry_id)},
+            manufacturer="Transmission",
+            name=client_name,
+        )
 
     @property
     def name(self):
         """Return the name of the sensor."""
-        return f"{self._client_name} {self._name}"
+        return f"{self._client_name} {self.entity_description.name_suffix}"
```

Comment:
With the new entity naming style, entities are no longer prefixed.

https://developers.home-assistant.io/docs/core/entity#entity-naming